Source/Core/Scripting/Python/Utils/fmt.h

```diff
@@ -0,0 +1,162 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+// This file lets you translate types to their corresponding python format strings.
+// See the python documentation for more information on those:
+// https://docs.python.org/3/c-api/arg.html
+//
+// Examples:
+//
+// Get the format string of a single type:
+//     const char* fmt_string = Py::fmt<float>;
+//     // returns "f"
+//
+// Get the format string of multiple types:
+//     std::string fmt_string = Py::fmts<float, PyObject*, int>;
+//     // returns "fOi"
+
+#pragma once
+
+#include <Python.h>
+#include <string>
+
+namespace Py
+{
+
+template <typename T>
+constexpr const char* GetPyFmt()
+{
+  // I guess automatic boolean parsing could be done with enough metaprogramming,
+  // but it seems to be very complicated.
+  static_assert((!std::is_same_v<T, bool>),
+                "The Python C API does not have a boolean type. "
+                "Consider using PyObject* and checking with PyObject_IsTrue(PyObject*)");
+  static_assert(false, R"(
```

Comment:
```suggestion
  static_assert(true, R"(
```
I changed this assert to be true at all times in my patch, so it would compile.
Just curious, what does this message mean and why as it set to always assert as false?

Reply:
It means that only code paths using template specializations are valid. If any usage of `GetPyFmt()` uses this generic implementation, it is an error because the type used does not have a corresponding python formatting character defined in one of the template specializations. I replaced this assert with `static_assert(sizeof(T) != sizeof(T), ...` instead on my fork